packages/Filters/src/components/Item/Item.js

```diff
@@ -0,0 +1,188 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import DeleteIcon from "@paprika/icon/lib/Trashbin";
+import Input from "@paprika/input";
+import Select from "@paprika/select";
+import DatePicker from "../DatePicker";
+import FilterContext from "../../context";
+import * as types from "../../types";
+import FilterPrefix from "../FilterPrefix";
+import * as sc from "./FilterItem.styles";
+
+const propTypes = {
+  columnId: PropTypes.string.isRequired,
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  index: PropTypes.number.isRequired,
+  onChangeFilter: PropTypes.func.isRequired,
+  onDeleteFilter: PropTypes.func.isRequired,
+  renderValueField: PropTypes.func,
+  rule: PropTypes.string.isRequired,
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
+};
+
+const defaultProps = {
+  id: null,
+  renderValueField: null,
+};
+
+function Item(props) {
+  const {
+    index,
+    onChangeFilter,
+    onDeleteFilter,
+    id,
+    columnId: selectedColumnId,
+    rule: selectedRule,
+    value,
+    renderValueField: renderCustomValueField,
+  } = props;
+  const { columns, data, filtersRef, onChangeOperator, operator, rulesByType } = React.useContext(FilterContext);
+  const I18n = useI18n();
+
+  const selectedColumnType = columns.find(({ id }) => id === selectedColumnId).type;
+
+  const selectOptions = React.useMemo(() => {
+    return selectedColumnType === types.columnTypes.SINGLE_SELECT
+      ? data.filter((obj, index, arr) => {
+          return arr.map(mapObj => mapObj[selectedColumnId]).indexOf(obj[selectedColumnId]) === index;
```

Comment:
Can you come up with a better name than `mapObj`?

`obj` isn't my favourite either, i usually go with `datum` (the singular of `data`), but that's not as big a deal